indico/modules/events/papers/controllers/display.py

```diff
@@ -16,29 +16,39 @@
 
 from __future__ import unicode_literals
 
-from flask import request, session
+from flask import request, session, render_template
 from werkzeug.exceptions import Forbidden
 
-from indico.modules.events.contributions.models.contributions import Contribution
-from indico.modules.events.papers.controllers.base import RHPapersBase
-from indico.modules.events.papers.forms import PaperSubmissionForm
-from indico.modules.events.papers.operations import create_paper_revision
-from indico.web.util import jsonify_form, jsonify_data
+from indico.modules.events.papers.controllers.base import RHPaperBase
+from indico.modules.events.papers.forms import PaperSubmissionForm, PaperCommentForm, build_review_form
+from indico.modules.events.papers.models.files import PaperFile
+from indico.modules.events.papers.models.reviews import PaperReviewType, PaperReview, PaperTypeProxy
+from indico.modules.events.papers.models.comments import PaperReviewComment
+from indico.modules.events.papers.operations import (create_paper_revision, create_review, create_comment,
+                                                     update_comment, delete_comment, update_review)
+from indico.modules.events.papers.views import WPDisplayPapersBase
+from indico.web.flask.templating import get_template_module
+from indico.web.util import jsonify_form, jsonify_data, jsonify
 
 
-class RHSubmitPaper(RHPapersBase):
-    normalize_url_spec = {
-        'locators': {
-            lambda self: self.contribution
-        }
-    }
+def render_paper_page(paper, view_class=None):
```

Comment:
Maybe they should. In my opinion, the rendering of stuff belongs precisely to the view.